tests/common/jwt/jwt_token.go

```diff
@@ -94,6 +94,28 @@ const (
 	// nolint: lll
 	TokenIssuer1WithNestedClaims2 = "eyJhbGciOiJSUzI1NiIsImtpZCI6InRUX3c5TFJOclk3d0phbEdzVFlTdDdydXRaaTg2R3Z5YzBFS1I0Q2FRQXciLCJ0eXAiOiJKV1QifQ.eyJleHAiOjQ3NTc2MDgwMTgsImlhdCI6MTYwNDAwODAxOCwiaXNzIjoidGVzdC1pc3N1ZXItMUBpc3Rpby5pbyIsIm5lc3RlZCI6eyJrZXkyIjoidmFsdWVDIiwibmVzdGVkLTIiOnsia2V5MiI6InZhbHVlQyJ9fSwic3ViIjoic3ViLTEifQ.dPBVaeOoMBFh9EFovHtmG2V-Bw106kUAV4GTPTVwoOaLxVUfAIpumrGX1fvEYLmtdwEkkL2CkkDdmKbsBgJBfnwg04oMeyHHrpSe5Y-8NJWGGxaBTzc7btWIbc1IPvj0vw_jIBeSqpXyW2Kd1Rc9kxK8rRuTdAeSK6oKcqg5d6kw-fC5PeiDlnuBH7F-xIaPYOHSxlKjC6TSP6-JYlQHw-uZ6rJmFwhUvc9B06dl3Xgx1W3PmYyt1e4kmHBPbwTgj1ckv8yev1JfIbciaNyQcgMf_eFwgQscHSeUY2e3v6DYAGHAQv0M6o_wweUlHfJP1juxUVA3UfpqC76NeiYipg"
 
+	// TokenIssuer1WithCollisionResistantName
+	// payload:
+	//{
+	//  "exp": 4837231657,
+	//  "iat": 1683631657,
+	//  "iss": "test-issuer-1@istio.io",
+	//  "sub": "sub-1",
+	//  "test-issuer-1@istio.io/nested": {
+	//    "key1": "valueC",
+	//    "nested-2": {
+	//      "key1": "valueC"
+	//    }
+	//  },
+	//  "test-issuer-1@istio.io/simple": "valueC"
+	//}
+	// Generated by: python3 security/tools/jwt/samples/gen-jwt.py tests/common/jwt/key.pem -jwks=tests/common/jwt/jwks.json
+	// --expire=3153600000 --iss=test-issuer-1@istio.io --sub=sub-1
+	// --claims=test-issuer-1@istio.io/simple:valueC
+	// --nestedkey=test-issuer-1@istio.io/nested --nestedclaim key1 valueC
+	// nolint: lll
+	TokenIssuer1WithCollisionResistantName = "eyJhbGciOiJSUzI1NiIsImtpZCI6InRUX3c5TFJOclk3d0phbEdzVFlTdDdydXRaaTg2R3Z5YzBFS1I0Q2FRQXciLCJ0eXAiOiJKV1QifQ.eyJleHAiOjQ4MzcyMzE2NTcsImlhdCI6MTY4MzYzMTY1NywiaXNzIjoidGVzdC1pc3N1ZXItMUBpc3Rpby5pbyIsInN1YiI6InN1Yi0xIiwidGVzdC1pc3N1ZXItMUBpc3Rpby5pby9uZXN0ZWQiOnsia2V5MSI6InZhbHVlQyIsIm5lc3RlZC0yIjp7ImtleTEiOiJ2YWx1ZUMifX0sInRlc3QtaXNzdWVyLTFAaXN0aW8uaW8vc2ltcGxlIjoidmFsdWVDIn0.kiXEekepka92nbKNZoPrjx4oeg5OI_ILROfZ9RR02tPsaXmVAEXzfsd5foend_seEru85oLXwiBM8uFJgn8hjE9oD4wXs-xGJHpgTogBeiuzIkrWN0o6iD03tYHFKCTUY8O5B5-9NoyYiEL5pNUqPd16VwfMRg_8qaOZPBVqES-mQED8xVL_UXOlkVQEsb2QUZD_9WfkzO1JRk8m1b6gxxxN2Its1jUB-F667yUQ5tUcYbuaus5yG5sWwpAJTXWzakC2Qq6NfB5hV2fwOsVSLUvH6B7sBq6dz6W03-xFH58j_aAAF0ceJ_LsX63hqZfUOeIcbKCRFoe6Ou8cw5sRtw"
```

Comment:
why do you need to create a new token, we can use the existing tokens? We already have a token with nested claims.

Reply:
> "test-issuer-1@istio.io/simple": "valueC"
> "test-issuer-1@istio.io/nested": {...}

The token's claim name contains special characters `@`, `.` and `/`, which are the scenarios we need to test.

Reply:
We don't have this kind of token now.